docs/reference/builder.md

```diff
@@ -180,21 +180,28 @@ that set `abc` to `bye`.
 
 ### .dockerignore file
 
-If a file named `.dockerignore` exists in the root of `PATH`, then Docker
-interprets it as a newline-separated list of exclusion patterns. Docker excludes
-files or directories relative to `PATH` that match these exclusion patterns. If
-there are any `.dockerignore` files in `PATH` subdirectories, Docker treats
-them as normal files. 
-
-Filepaths in `.dockerignore` are absolute with the current directory as the
-root. Wildcards are allowed but the search is not recursive. Globbing (file name
-expansion) is done using Go's
-[filepath.Match](http://golang.org/pkg/path/filepath#Match) rules.
-
-You can specify exceptions to exclusion rules. To do this, simply prefix a
-pattern with an `!` (exclamation mark) in the same way you would in a
-`.gitignore` file. Currently there is no support for regular expressions.
-Formats like `[^temp*]` are ignored. 
+If a file named `.dockerignore` exists at the root of the source
+repository, Docker interprets it as a newline-separated list of exclusion
+patterns. Docker excludes from the build context files or directories whose
+paths relative to the root of the source repository match these exclusion
+patterns. If there are any files named `.dockerignore` elsewhere in the
+source repository, Docker treats them as normal files.
+
+Matching is done using Go's
+[filepath.Match](http://golang.org/pkg/path/filepath#Match) rules,
+which are similar to those of Unix shells (but also different).  A
+preprocessing step removes leading and trailing whitespace and
+eliminates `.` and `..` elements using Go's
+[filepath.Clean](http://golang.org/pkg/path/filepath/#Clean).  Lines
+that are blank after preprocessing are ignored.
+
+Sometimes, you may want to exclude all files matching a certain pattern
```

Comment:
"You can specify exceptions to exclusion rules." is easier to parse than what you replaced it with.  I see where you are going though so.

----->

The exclusion pattern syntax uses Go's [filepath.Match](http://golang.org/pkg/path/filepath#Match) rules, which are similar to those of Unix shells. A preprocessing step removes leading and trailing whitespace and eliminates `.` and `..` elements using Go's[filepath.Clean(http://golang.org/pkg/path/filepath/#Clean). Lines that are blank after preprocessing are ignored.

You can specify an exception to exclusion pattern. To do this, add a second, exception pattern prefixed by  a `!` (exclamation mark) after the exclusion pattern. This `.dockerignore` file excludes all Markdown files from the build except the `LICENSE.md` file.

```
*.md
!LICENSE.md
```    

More generally, a file that matches at least one pattern in the
`.dockerignore` file is included in the build context if and only if
the *last* pattern is prefixed with `!`  (exclamation mark).